modules/swagger-codegen/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -42,8 +42,8 @@ class ObjectSerializer
     {
         if (is_scalar($data) || null === $data) {
             return $data;
-        } elseif ($data instanceof \DateTime) {
-            return ($format === 'date') ? $data->format('Y-m-d') : $data->format(\DateTime::ATOM);
+        } elseif ($data instanceof \DateTimeInterface) {
```

Comment:
Not sure about the BC impact as `DateTimeInterface` was added in PHP 5.5 whereas `DateTime` was added in PHP 5.2
If impact matters for this case, we can probably add an or condition: `$data instanceof \DateTime || $data instanceof \DateTimeInterface`